openedx/core/djangoapps/content_libraries/rest_api/serializers.py

```diff
@@ -426,3 +434,101 @@ class LibraryBackupTaskStatusSerializer(serializers.Serializer):
     """
     state = serializers.CharField()
     url = serializers.FileField(source='file', allow_null=True, use_url=True)
+
+
+class LibraryRestoreFileSerializer(serializers.Serializer):
+    """
+    Serializer for restoring a library from a backup file.
+    """
+    # input only fields
+    file = serializers.FileField(write_only=True, help_text="A ZIP file containing a library backup.")
+
+    # output only fields
+    task_id = serializers.UUIDField(read_only=True)
+
+    def validate_file(self, value):
+        """
+        Validate that the uploaded file is a ZIP file.
+        """
+        if value.content_type != 'application/zip':
+            raise serializers.ValidationError("Only ZIP files are allowed.")
+        return value
+
+
+class LibraryRestoreTaskRequestSerializer(serializers.Serializer):
+    """
+    Serializer for requesting the status of a library restore task.
+    """
+    task_id = serializers.UUIDField(write_only=True, help_text="The ID of the restore task to check.")
+
+
+class RestoreSuccessDataSerializer(serializers.Serializer):
+    """
+    Serializer for the data returned upon successful restoration of a library.
+    """
+    learning_package_id = serializers.IntegerField(source="lp_restored_data.id")
+    title = serializers.CharField(source="lp_restored_data.title")
+    org = serializers.CharField(source="lp_restored_data.archive_org_key")
+    slug = serializers.CharField(source="lp_restored_data.archive_slug")
+    key = serializers.CharField(source="lp_restored_data.key")
+    archive_key = serializers.CharField(source="lp_restored_data.archive_lp_key")
```

Comment:
Please add a short comment on the difference between the two and why they would be different.